HopsanGenerator/templates/fmu_hopsan.cpp

```diff
@@ -138,45 +149,88 @@ void hopsan_get_message(hopsan_message_callback_t message_callback, void* userSt
 
 double hopsan_get_real(int vr)
 {
-    return (*dataPtrs[vr]);
+    if(vr < sizeof(dataPtrs)/sizeof(double*)) {
+        return (*dataPtrs[vr]);
+    }
+    if(realParametersMap.count(vr)) {
+        hopsan::HString value;
+        spCoreComponentSystem->getParameterValue(realParametersMap[vr],value);
+        bool ok;
+        return value.toDouble(&ok);
+    }
+    return 0;
```

Comment:
Maybe use -1 for "error value" for doubles as well, (it is used for ints below) and 0 could be confused with a typical default value.

Reply:
Fixed.